yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -136,32 +136,90 @@ impl ToTokens for HtmlTag {
+                    {
+                        let __yew_value = ::std::option::Option::as_ref(&(#value)).map(::std::string::ToString::to_string);
+                        (::std::string::String::from(#label_str), __yew_value)
+                    }
+                }
+            } else {
+                quote_spanned! {value.span()=>
+                    {
+                        (::std::string::String::from(#label_str), ::std::option::Option::Some(::std::string::ToString::to_string(&#value)))
+                    }
                 }
             }
         });
+        let set_booleans = booleans.iter().map(
+            |TagAttribute {
+                 label,
+                 question_mark: _,
+                 value,
+             }| {
+                let label_str = label.to_string();
+                quote_spanned! {value.span()=>
+                    {
+                        if #value {
+                            #vtag.add_attribute(#label_str, &#label_str);
+                        }
+                    }
+                }
+            },
+        );
         let set_kind = kind.iter().map(|kind| {
-            quote_spanned! {kind.span()=> #vtag.set_kind(&(#kind)); }
+            let value = &kind.value;
+            if kind.question_mark.is_some() {
+                quote_spanned! {value.span()=>
+                    {
+                        if let ::std::option::Option::Some(__yew_kind) = ::std::option::Option::as_ref(&(#value)) {
+                            #vtag.set_kind(__yew_kind)
+                        }
+                    }
+                }
+            } else {
+                quote_spanned! {value.span()=> #vtag.set_kind(&(#value)); }
+            }
         });
         let set_value = value.iter().map(|value| {
-            quote_spanned! {value.span()=> #vtag.set_value(&(#value)); }
+            let value_value = &value.value;
+            if value.question_mark.is_some() {
+                quote_spanned! {value_value.span()=>
+                    {
+                        if let ::std::option::Option::Some(__yew_value) = ::std::option::Option::as_ref(&(#value_value)) {
+                            #vtag.set_value(__yew_value);
+                        }
+                    }
+                }
+            } else {
+                quote_spanned! {value_value.span()=> #vtag.set_value(&(#value_value)); }
+            }
         });
         let add_href = href.iter().map(|href| {
-            quote_spanned! {href.span()=>
-                let __yew_href: ::yew::html::Href = (#href).into();
-                #vtag.add_attribute("href", &__yew_href);
+            let value = &href.value;
+            if href.question_mark.is_some() {
+                quote_spanned! {value.span()=>
+                    {
+                        if let ::std::option::Option::Some(__yew_href) = #value {
+                            let __yew_href: ::yew::html::Href = __yew_href.into();
```

Comment:
Also something unrelated to the PR but I think we try to be more consistent.
```suggestion
                            let __yew_href = ::yew::html::Href::from(__yew_href);
```